jenkins/bootstrap.py

```diff
@@ -735,19 +737,44 @@ def gubernator_uri(paths):
     return job
 
 
+@contextlib.contextmanager
+def choose_ssh_key(ssh):
+    """Creates a script for GIT_SSH that uses -i ssh if set."""
+    if not ssh:  # Nothing to do
+        yield
+        return
+
+    with tempfile.NamedTemporaryFile(prefix='ssh', delete=False) as fp:
+        fp.write('#!/bin/sh\nssh -i \'%s\' -F /dev/null "${@}"\n' % ssh)
+    try:
+        os.chmod(fp.name, 0500)
+        had = 'GIT_SSH' in os.environ
```

Comment:
maybe also set $GIT_SSH_COMMAND?
ref http://man7.org/linux/man-pages/man1/git.1.html
```
       GIT_SSH, GIT_SSH_COMMAND
           If either of these environment variables is set then git fetch
           and git push will use the specified command instead of ssh when
           they need to connect to a remote system. The command will be
           given exactly two or four arguments: the username@host (or just
           host) from the URL and the shell command to execute on that
           remote system, optionally preceded by -p (literally) and the port
           from the URL when it specifies something other than the default
           SSH port.

           $GIT_SSH_COMMAND takes precedence over $GIT_SSH, and is
           interpreted by the shell, which allows additional arguments to be
           included.  $GIT_SSH on the other hand must be just the path to a
           program (which can be a wrapper shell script, if additional
           arguments are needed).

           Usually it is easier to configure any desired options through
           your personal .ssh/config file. Please consult your ssh
           documentation for further details.
```